### DISCUSSION TITLE
RFC: Mint as a metalanguage for React Native

### DISCUSSION THREAD

**Discussion by lukepighetti** [mint-lang/mint#284] [Language Development]
I've been thinking about this a lot. Mint has some features that really make it stand out in the world of front end programming. We use Flutter a lot at our mobile app agency and we'd love to use Mint as a metalanguage for that ecosystem, but that seems very unlikely. However, it seems like `Mint` could be a metalanguage for React Native, including React Native Web. `style` could use the React Native style properties and Mint could target web & mobile.

Any thoughts?

**Comment by s0kil** [comment]
Related To: https://github.com/mint-lang/mint/issues/111

**Comment by gdotdesign** [comment]
I think the architecture of the project would allow compiling down to other targets:
- the AST and type checker can be reused as is
- `Html` probably needs to be renamed to something else
- if the target is not JavaScript then the core library needs to be migrated as well
- how styling works needs to be rethought 
- much tooling is needed 

It is certainly possible, someone just need to put the work in :slightly_smiling_face:
